src/overlays/actors/ovl_En_Mag/z_en_mag.c

```diff
@@ -28,28 +97,878 @@ const ActorInit En_Mag_InitVars = {
+                                gSaveContext.sceneSetupIndex = 0;
+                            }
+                            this->unk11F54 = 15;
+                            this->unk11F56 = 25;
+                            this->state = MAG_STATE_FADE_OUT;
+                        }
+                    } else {
+                        sInputDelayTimer--;
+                    }
+                    break;
+
+                case MAG_STATE_FADE_OUT:
+                    TIMED_STEP_DOWN_TO(this->appearEffectAlpha, 0, this->fadeOutTimer, step);
+                    TIMED_STEP_DOWN_TO(this->majorasMaskAlpha, 0, this->fadeOutTimer, step);
+                    TIMED_STEP_DOWN_TO(this->displayEffectAlpha, 0, this->fadeOutTimer, step);
+                    TIMED_STEP_DOWN_TO(this->mainTitleAlpha, 0, this->fadeOutTimer, step);
+                    TIMED_STEP_DOWN_TO(this->subtitleAlpha, 0, this->fadeOutTimer, step);
+                    TIMED_STEP_DOWN_TO(this->copyrightAlpha, 0, this->fadeOutTimer, step);
+
+                    this->fadeOutTimer--;
+                    if (this->fadeOutTimer == 0) {
+                        this->appearEffectAlpha = 0;
+                        this->majorasMaskAlpha = 0;
+                        this->mainTitleAlpha = 0;
+                        this->subtitleAlpha = 0;
+                        this->displayEffectAlpha = 0;
+                        this->copyrightAlpha = 0;
+                        this->state = MAG_STATE_POST_DISPLAY;
+                    }
+                    break;
+            }
+
+            // Appear fully immediately if called during fade-in states.
+            if ((this->state > MAG_STATE_INITIAL) && (this->state < MAG_STATE_CALLED)) {
+                if (CHECK_BTN_ALL(CONTROLLER1(globalCtx)->press.button, BTN_START) ||
+                    CHECK_BTN_ALL(CONTROLLER1(globalCtx)->press.button, BTN_A) ||
+                    CHECK_BTN_ALL(CONTROLLER1(globalCtx)->press.button, BTN_B)) {
+                    play_sound(NA_SE_SY_PIECE_OF_HEART);
+                    this->state = MAG_STATE_CALLED;
+                }
+            }
+        }
+    }
+
+    if (this->state == MAG_STATE_INITIAL) {
+        if (EnvFlags_Get(globalCtx, 3)) {
+            this->unk11F02 = 40;
+            this->state = MAG_STATE_FADE_IN_MASK_EFFECTS;
+        }
+    } else if (this->state < MAG_STATE_FADE_OUT) {
+        if (EnvFlags_Get(globalCtx, 4)) {
+            this->state = MAG_STATE_FADE_OUT;
+        }
+    }
+}
+
+/**
+ * Draws an i8 texture.
+ *
+ * @param gfxp Pointer to current displaylist.
+ * @param texture Texture to draw.
+ * @param texWidth Width of the texture.
+ * @param texHeight Height of the texture.
+ * @param rectLeft X coordinate of the top-left of the draw position.
+ * @param rectTop Y coordinate of the top-left of the draw position.
+ */
+void EnMag_DrawTextureI8(Gfx** gfxp, void* texture, s16 texWidth, s16 texHeight, s16 rectLeft, s16 rectTop) {
```

Comment:
```suggestion
void EnMag_DrawTextureI8(Gfx** gfxp, TexturePtr texture, s16 texWidth, s16 texHeight, s16 rectLeft, s16 rectTop) {
```
More of a thought/question rather than suggestion. Applies elsewhere.

Reply:
I think that makes sense. The functions where it matters treat them as a byte array anyway.